tests/expose_test.go

```diff
@@ -409,4 +410,117 @@ var _ = Describe("Expose", func() {
 			})
 		})
 	})
+
+	Context("Verify that an exposed service on a VM survives after restarting the VM.", func() {
+		const servicePort = "27017"
+		const serviceName = "cluster-ip-vm-restart"
+		var vm_obj *v1.VirtualMachine
+
+		tests.BeforeAll(func() {
+			By("Creating an VM object")
+			template := tests.NewRandomVMIWithEphemeralDiskAndUserdata(tests.ContainerDiskFor(tests.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
+			template.Labels = map[string]string{"expose": "vm"}
+			vm_obj = NewRandomVirtualMachine(template, false)
+
+			By("Creating the VM")
+			_, err := virtClient.VirtualMachine(tests.NamespaceTestDefault).Create(vm_obj)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Starting the VMI.")
+			virtctl := tests.NewRepeatableVirtctlCommand("start", "--namespace", vm_obj.Namespace, vm_obj.Name)
+			err = virtctl()
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Verifying the VMI is running.")
+			Eventually(func() bool {
+				vm_obj, err = virtClient.VirtualMachine(vm_obj.Namespace).Get(vm_obj.Name, &k8smetav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return vm_obj.Status.Ready
+			}, 120*time.Second, 1*time.Second).Should(BeTrue())
+
+			By("Exposing a service on the VM using virtctl")
+			virtctl = tests.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "virtualmachine", "--namespace",
+				vm_obj.Namespace, vm_obj.Name, "--port", servicePort, "--name", serviceName, "--target-port", strconv.Itoa(testPort))
+			err = virtctl()
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Getting the running VMI")
+			var vmi *v1.VirtualMachineInstance
+			Eventually(func() bool {
+				vmi, err = virtClient.VirtualMachineInstance(vm_obj.Namespace).Get(vm_obj.Name, &k8smetav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return vmi.Status.Phase == v1.Running
+			}, 120*time.Second, 1*time.Second).Should(BeTrue())
+
+			generateHelloWorldServer(vmi, virtClient, testPort, "tcp")
+		})
+
+		Context("Verify the exposed service is functional before and after VM restart.", func() {
+			It("Connect to exposed ClusterIP service.", func() {
```

Comment:
This should be one case, you should have one 'It' and everything should be 'By'.
'It' order is not generated .